BackEnd/src/models/generic.model.js

```diff
@@ -0,0 +1,139 @@
+        })
+    }
+    static async getID(id){
+        try {
+            return await databaseConn.query(`SELECT * FROM ${this.tableName} WHERE ${this.fields[0]} = ?`,[String(id)],{saveAsPrepared:true})
+        } catch (error) {
+            console.error(error)
+            throw(error)
+        }
+    }
+    static getAllSync(res){
+        databaseConnSync.getConnection((err,conn)=>{
+            if(err){
+                console.error(err)
+                if(err.errno !== -4078) conn.release()
+            }
+            else conn.query(`SELECT * FROM ${this.tableName}`,(error,results,fields)=>{
+                conn.release()
+                if(error) console.error(error)
+                else res(results)
+            })
+        })
+    }
+    static async getAll(){
+        try {
+            return await databaseConn.query(`SELECT * FROM ${this.tableName}`)
+        } catch (error) {
+            console.error(error)
+            throw(error)
+        }
+    }
+    static getFieldsSync(res){
+        databaseConnSync.getConnection((err,conn)=>{
+            if(err){
+                console.error(err)
+                if(err.errno !== -4078) conn.release()
+            }
+            else conn.query(`SELECT * FROM ${this.tableName}`,(error,results,fields)=>{
+                conn.release()
+                if(error) console.error(error)
+                else res(fields.map(a=>a.name))
+            })
+        })
+    }
+    static async getFields(){
+        try {
+            let results = await databaseConn.query(`DESCRIBE ${this.tableName}`)
+            return results.map(a=>a.Field)
+        } catch (error) {
+            console.error(error)
+            throw(error)
+        }
+    }
+    createSync(res,error){
+        const values = Object.values(this).map(value=>value ?? null).slice(2)
+        const fields = this.table.fields.slice(1)
+        databaseConnSync.getConnection((err,conn)=>{
+            if(err){
+                error(err)
+                if(err.errno !== -4078) conn.release()
+            }
+            else {
+                conn.execute(`INSERT INTO ${this.table.name}(${fields.join(`, `)}) VALUES(${values.map(() => '?').join(', ')})`,values,(err, results)=>{
+                    conn.release()
+                    if(err){
+                        error(err)
+                    }
+                    else {
+                        console.log(`${this.table.name} ID ${results.insertId} has been successfully inserted.`)
+                        res(results)
+                    }
+                })
+            }
+        })
+    }
+    async create(){
+        try {
+            const values = Object.values(this).map(value=>value ?? null).slice(2)
+            const fields = this.table.fields.slice(1)
+            return await databaseConn.query(`INSERT INTO ${this.table.name}(${fields.join(`, `)}) VALUES(${values.map(() => '?').join(', ')})`,values,{saveAsPrepared:true})
+        } catch (error) {
```

Comment:
Remove this line and change dbconn in parameters to conn when transaction is set up for all the create methods